autoload/sy.vim

```diff
@@ -76,50 +67,23 @@ function! sy#start(...) abort
 endfunction
 
 " #stop {{{1
-function! sy#stop(bufnr) abort
-  if empty(getbufvar(a:bufnr, 'sy'))
-    return
-  endif
-  call sy#sign#remove_all_signs(a:bufnr)
-endfunction
-
-" #enable {{{1
-function! sy#enable() abort
-  if !exists('b:sy')
-    call sy#start()
-    return
-  endif
-
-  if !b:sy.active
-    let b:sy.active = 1
-    let b:sy.retry  = 1
-    call sy#start()
-  endif
-endfunction
-
-" #disable {{{1
-function! sy#disable(...) abort
-  let sy = getbufvar(a:0 ? a:1 : bufnr(''), 'sy')
-
-  if !empty(sy) && sy.active
-    call sy#stop(sy.buffer)
-    let b:sy.active = 0
-    let b:sy.stats = [-1, -1, -1]
-  endif
+function! sy#stop(...) abort
+  let bufnr = bufnr('')
+  if empty(getbufvar(a:0 ? a:1 : bufnr, 'sy')) | return | endif
+  call sy#sign#remove_all_signs(bufnr)
+  " TODO: Can't unset autocmds in another buffer.
```

Comment:
Like `autocmd! signify * <buffer=33>`?

Reply:
I think we want per-buffer behavior here anyway, so I'm not sure what the TODO is for. If we want a `SignifyEnableAll` or `SignifyDisableAll` command, we'd need to place signs or remove signs in all buffers. Doing so for a specific buffer which is not the current one seems a bit unnecessarily complex.

Reply:
Uh.. exactly. Thanks!

Reply:
This is needed because `sy#stop()` could be called from a callback handler. Due to async nature of this plugin, the user isn't necessarily in the same buffer anymore as when the `sy#start()` was called.